### DISCUSSION TITLE
The differences between pFBA and normal FBA

### DISCUSSION THREAD

**Discussion by complicatedZ** [opencobra/cobrapy#1353] [Q&A]
Dear developers,
Compared with normal FBA, we know that pFBA minimizes the total sum of flux at the same time. And as I calculate the total some of flux optimized in two different ways, the results are really close as below.
![1693985842870](https://github.com/opencobra/cobrapy/assets/112108352/a05dffbb-539e-4c3d-920d-dc93294edd5c)
I wonder it's a coincidence or the normal FBA optimization designed to minimize the sum of flux **to some extent**, because due to many loops existed in the model, the sum of flux could be very high if you just want to maxmize the growth rate.
Best Regards.

**Comment by Midnighter** [comment]
Is that the small textbook model or a genome-scale model? That may have something to do with it.

Additionally, there is probably solution caching going on. You first compute the pFBA which uses the optimal value from a normal FBA, sets that as a constraint, then minimizes the fluxes. So when you next perform a normal FBA, the solver may detect that the current state is already an optimal solution and does not change the flux distribution at all. It's possible that you will see a larger difference when you first do a normal FBA and then a pFBA.

**Reply by complicatedZ** [reply]
Thank you for your reply! By the way, the model is a genome-scale model including more than 2,000 reactions, is there something special?
